File: python/knapsack/knapsack.py
import itertools
def maximum_value(maximum_weight, items):
    value = 0
    for i in range(1, len(items) + 1):
        combos = itertools.combinations(items, i)
        for c in combos:
            weight = sum(x['weight'] for x in c)
            if weight <= maximum_weight:
                v = sum(x['value'] for x in c)
                if v > value: value = v
    return value

File: ruby/knapsack/knapsack.rb
Item = Struct.new(:weight, :value, keyword_init: true)

class Knapsack
  def initialize(max_weight)
    @max_weight = max_weight
  end

  def max_value(items)
    (1..items.size).flat_map { |size|
      items.combination(size).map { |combo|
        combo.sum(&:weight) <= @max_weight ? combo.sum(&:value) : 0
      }
    }.max || 0
  end
end
